### PR TITLE
Make use of `na_matches` argument consistent, and document all in generics' doc

### PR DESCRIPTION
**Pull request by moodymudskipper**
Addresses https://github.com/tidyverse/dplyr/issues/4225

The methods' help file seemed outdated (was using `tbl_df` in the examples) and not adding value compared to the generics' help file, and didn't even list all generics.

Argument `na_matches` was not listed for generics not for `data.frame` methods while it was fully supported as it was passed to `*.tbl_df` through dots.

This PR makes things consistent by making the argument explicit everywhere relevant, adding the description of the na_matches argument in the main help file, and removing the help file of methods and the sentence that referenced them at the start of the generics' help file.

(help file hereby meaning including roxygen comments)

### CODE REVIEW

File: R/join.r [hunk 1]
```diff
@@ -88,6 +87,13 @@
 #' @param keep If `TRUE` the by columns are kept in the nesting joins.
 #' @param ... other parameters passed onto methods, for instance, `na_matches`
 #'   to control how `NA` values are matched.  See \link{join.tbl_df} for more.
```
Comment (romainfrancois) [inline R/join.r]:
```suggestion
#'   to control how `NA` values are matched.  See [join.tbl_df()] for more.
```

Comment (moodymudskipper) [reply R/join.r]: Actually the part "See [join.tbl_df()] for more." should go as the help file doesn't exist anymore, sorry for missing that.

Comment (romainfrancois) [reply R/join.r]: Makes sense. Let's see if that works.

File: R/join.r [hunk 2]
```diff
@@ -87,7 +86,14 @@
 #' @param name the name of the list column nesting joins create. If `NULL` the name of `y` is used.
 #' @param keep If `TRUE` the by columns are kept in the nesting joins.
 #' @param ... other parameters passed onto methods, for instance, `na_matches`
-#'   to control how `NA` values are matched.  See \link{join.tbl_df} for more.
+#'   to control how `NA` values are matched.  See [join.tbl_df()] for more.
```
Comment (romainfrancois) [inline R/join.r]:
```suggestion
#'   to control how `NA` values are matched.
```

**Comment by moodymudskipper** [comment]
@romainfrancois please tell me if you need further action from me on this I'm still largely confused about the way github collaboration works.

**Comment by hadley** [comment]
I don't think this is a good approach. We'll need to take a look when we're next working on joins.

**Comment by moodymudskipper** [comment]
Alright, if I'm in a position to help I'll be happy to.

**Pull request closed without merging**